### PR TITLE
Add TDX tests

### PR DESCRIPTION
**Pull request by asvrada**
Exactly the same as #4727

### CODE REVIEW

**Comment by mingweishih** [comment]
bors try

**Comment by asvrada** [comment]
bors try

**Comment by asvrada** [comment]
bors try-

**Comment by asvrada** [comment]
bors try

**Review by CyanDevs — APPROVED** [review]
LGTM and this has been merged into 0.19.x with tests passing in CI when including the appropriate environment variables in this PR: https://github.com/openenclave/openenclave/pull/4753

For the commented out target and test - when can those be re-enabled?

**Review by radhikaj — APPROVED** [review]

File: tests/secure_verify/host/evidence.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include <openenclave/bits/evidence.h>
 #include <openenclave/host.h>
 
-#include "../../../../host/sgx/platformquoteprovider.h"
+#include "../../../host/sgx/platformquoteprovider.h"
```
Comment (asvrada) [inline tests/secure_verify/host/evidence.h:10]: ## Why we modify this line
It's falling the pipeline `PR Testing / Ubuntu 1804 Cross Platform Build`

Previously, `../../../../host/sgx/platformquoteprovider.h` only work if build directory is 1 level deep, like `./build/`

> Then why on Windows it builds successfully where the build directory is 2 levels deep, like `./build/debug/`?

**Review by CyanDevs — APPROVED** [review]

**Review by mingweishih — APPROVED** [review]

**Pull request merged**